model/src/main/kotlin/Hash.kt

```diff
@@ -97,6 +97,11 @@ data class Hash(
 
         return algorithm.calculate(file).equals(value, ignoreCase = true)
     }
+
+    /**
+     * Verify that the provided [hash] matches this hash.
+     */
+    fun verify(hash: Hash): Boolean = algorithm == hash.algorithm && value.equals(hash.value, ignoreCase = true)
```

Comment:
Nit: Maybe rename the `hash` parameter to `other`?

Reply:
I thought about it, but it is also nice if the options in code completion show up as `file: File`, `hash: Hash`.